chall08/pcuadrad.py

```diff
@@ -0,0 +1,185 @@
+        '''Allows robot moves.'''
+        self.active = True
+
+    def stopEngine(self):
+        '''Stop robot move. '''
+        self.active = False
+
+class MineField:
+    def __init__(self):
+        self.mines = []
+        self.map = []
+        self.invalidMap = True
+        self.exitMark = False
+        self.robotMark = False
+
+    def setMine(self, posX, posY):
+        '''Creates a new Mine object with the x and y position.'''
+        self.mines.append(Mine(posX, posY))
+
+    def setExit(self, posX, posY):
+        '''Creates a new Exit object with the x and y position.'''
+        self.exit = Exit(posX, posY)
+        self.exitMark = True
+    
+    def setRobot(self, posX, posY):
+        '''Creates a new Robot object with the x and y position.'''
+        self.robot = Robot(posX, posY)
+        self.robotMark = True
+
+    def robotFoundExit(self) -> bool:
+        '''Return True if the robot found the exit and is already stopped.'''
+        if self.robot.exitFound(self.exit) and not self.robot.active:
+            return True
+        return False
+
+    def verifyLine(self, line: str, pos: int, totalPos: int) -> bool:
+        '''Verify a line of the map. Check that it is closed at the begining\
+            and in the end. It also creates the objects that are found, and\
+                if the exit is found or the robot more than once returns failure.'''
+        if pos == 0 or pos + 2 == totalPos:
+            for square, posy in zip(line, range(len(line))):
+                if square == '+':
+                    pass
+                elif square == 'E':
+                    if not self.exitMark:
+                        self.setExit(pos, posy)
+                    else:
+                        return False
+                else:
+                    return False
+            return True
+        else:
+            limit = 0
+            for square, posy in zip(line, range(len(line))):
+                if (square == '+' and (posy == 0 or posy + 1 == len(line))):
+                    limit += 1
+                elif square == '0':
+                    pass
+                elif square == 'E':
+                    limit += 1
+                    if not self.exitMark:
+                        self.setExit(pos, posy)
+                    else:
+                        return False
+                elif square == '*':
+                    self.setMine(pos, posy)
+                elif square == 'M':
+                    if not self.robotMark:
+                        self.setRobot(pos, posy)
+                    else:
+                        return False
+                else:
+                    return False
+        return True if limit == 2 else False
+
+    def constructMap(self, mapi):
+        '''Create the map in a two-dimensional array and check for errors.'''
+        self.invalidMap = False
+        for line, linePos in zip(mapi, range(len(mapi) - 1)):
+            if linePos > 0 and len(line) != len(lastLine):
```

Comment:
I know this won't happen normally, but reading the variable `lastLine` here before you initially assign it in line 126, is not good practice.